server/src/main/java/keywhiz/service/daos/SecretDAO.java

```diff
@@ -135,16 +135,15 @@ public long createSecret(String name, String encryptedSecret,
     SecretSeriesDAO secretSeriesDAO = secretSeriesDAOFactory.using(dslContext.configuration());
 
     Optional<SecretSeries> secretSeries = secretSeriesDAO.getSecretSeriesByName(name);
-    if (!secretSeries.isPresent()) {
-      return Optional.empty();
-    }
+    if (secretSeries.isPresent() && secretSeries.get().currentVersion().isPresent()) {
+      Optional<SecretContent> secretContent = secretContentDAO.getSecretContentBySecretIdOne(secretSeries.get().id());
+      if (!secretContent.isPresent()) {
+        return Optional.empty();
+      }
 
-    Optional<SecretContent> secretContent = secretContentDAO.getSecretContentBySecretIdOne(secretSeries.get().id());
-    if (!secretContent.isPresent()) {
-      return Optional.empty();
+      return Optional.of(SecretSeriesAndContent.of(secretSeries.get(), secretContent.get()));
     }
-
-    return Optional.of(SecretSeriesAndContent.of(secretSeries.get(), secretContent.get()));
+    return Optional.empty();
```

Comment:
if the row exists in the secrets table and the row's current is not null, then we treat the secret as present. Otherwise, the code should behave as if the secret doesn't exist.